eventuals/static-thread-pool.cc

```diff
@@ -151,14 +150,13 @@ void StaticThreadPool::Submit(Callback<> callback, Context* context) {
 ////////////////////////////////////////////////////////////////////////
 
 bool StaticThreadPool::Continuable(Context* context) {
-  auto* waiter = static_cast<Waiter*>(CHECK_NOTNULL(context));
-
-  CHECK(!waiter->waiting) << waiter->name();
-  CHECK(waiter->next == nullptr) << waiter->name();
+  CHECK(!context->blocked_) << context->name();
+  CHECK(context->next_ == nullptr) << context->name();
 
-  auto& pinned = waiter->requirements()->pinned;
+  auto& pinned =
+      static_cast<StaticThreadPool::Requirements*>(context->data_)->pinned;
```

Comment:
Let's be more explicit here and everywhere else (in other files too):
```suggestion
  auto* requirements = static_cast<StaticThreadPool::Requirements*>(context->data);
  auto& pinned = requirements->pinned;
```